### PR TITLE
Fix #596: Packaging now fails if no weavers are included in package

### LINKED ISSUE #596: FodyPackaging generates package without weavers
Du to wrong build order FodyPackaging generated the package before the weaver was built.
It created an invalid package without weavers, but it should have failed.

### PR DESCRIPTION
**Pull request by tom-englert**
Fix #596: Packaging now fails if no weavers are included in package

### CODE REVIEW

File: FodyPackaging/build/FodyPackaging.targets [hunk 1]
```diff
@@ -9,19 +9,19 @@
...
-      <Output TaskParameter="Include" ItemName="TfmSpecificPackageFile" />
-    </CreateItem>
-    <CreateItem Include="$(WeaverPropsTempFile)" AdditionalMetadata="PackagePath=build">
-      <Output TaskParameter="Include" ItemName="TfmSpecificPackageFile" />
-    </CreateItem>
+    <PropertyGroup>
+      <AllWeaverFilesIncluded>@(NetClassicFilesToInclude)@(NetStandardFilesToInclude)</AllWeaverFilesIncluded>
+    </PropertyGroup>
+
+    <Error Text="FodyPackaging: No weavers found to include in package. Maybe the build order is wrong?" Condition="'$(AllWeaverFilesIncluded)' == ''" />
```
Comment (ltrzesniewski) [inline FodyPackaging/build/FodyPackaging.targets]: Wouldn't it be better to raise an error if any one of the weavers is missing? I mean when either netstandard or netclassic is missing.
Having one but not the other could happen if you build projects in parallel.

Comment (tom-englert) [reply FodyPackaging/build/FodyPackaging.targets]: Good point, I'll gonna change this.

**Comment by SimonCropp** [comment]
Thanks. This should now be deployed pending a nuget update

**Comment by ltrzesniewski** [comment]
This can unfortunately cause false positives.

Here's what I was doing previously (simplified):

```
msbuild src\InlineIL.sln /t:Restore 
msbuild src\InlineIL.sln /t:Rebuild
msbuild src\InlineIL\InlineIL.csproj /t:Pack
```

So, I first builded everything (with `GeneratePackageOnBuild` set to `false`), then I explicitly created the NuGet package. This always produced valid packages.

This change triggered the new error on the `Rebuild` step. 😢 

My current workaround is this (again, simplified):

```
msbuild src\InlineIL.sln /t:Restore
msbuild src\InlineIL.Fody\InlineIL.Fody.csproj /t:Build
msbuild src\InlineIL.sln /t:Build
msbuild src\InlineIL\InlineIL.csproj /t:Pack
```

I need to explicitly build the weaver project first. I couldn't find any other way to make this work 😢 

 - Setting project dependencies on the solution works in VS but fails on the command line
 - Adding a project reference with `ReferenceOutputAssembly="false" SkipGetTargetFrameworkProperties="true" GlobalPropertiesToRemove="TargetFramework"` as explained in https://github.com/Microsoft/msbuild/issues/3843 causes the same errors as if I had set project dependencies in VS
 - Using `dotnet build` instead of `msbuild` doesn't change anything (understandably)

So I'm basically experiencing #505 once again here 😞 

I have no clue how you guys manage to get reliable builds in your weavers without doing any of this. I couldn't find any hidden tricks. Am I missing something obvious here?

I'll submit a PR to postpone the check until the actual packing takes place.

**Comment by ltrzesniewski** [comment]
Disregard my comment above. Turns out I haven't noticed that `FodyPackaging` sets `GeneratePackageOnBuild` to true.

**Pull request merged**